src/test/groovy/com/cloudbees/groovy/cps/CpsTransformerTest.groovy

```diff
@@ -612,6 +612,13 @@ class CpsTransformerTest extends AbstractGroovyCpsTest {
         ''') == [-3,2];
     }
 
+    @Issue("JENKINS-31484")
+    @Test
+    void fieldViaGetter() {
+        assert evalCPS('class C {private int x = 33}; new C().x') == 33
+        assert evalCPS('class C {private int x = 33; int getX() {2 * this.@x}}; new C().x') == 66
+        assert evalCPS('class C {private int x = 33; int getX() {2 * x}}; new C().x') == 66
```

Comment:
Originally the first two were fine, but the third would go into an endless loop. Instrumenting `Next.step` indicated that accessing `x` was calling the property, which then called `getX()`  again.